azurerm/internal/services/storage/resource_arm_storage_account.go

```diff
@@ -1131,10 +1137,12 @@ func resourceArmStorageAccountRead(d *schema.ResourceData, meta interface{}) err
 		d.Set("enable_https_traffic_only", props.EnableHTTPSTrafficOnly)
 		d.Set("is_hns_enabled", props.IsHnsEnabled)
 		d.Set("allow_blob_public_access", props.AllowBlobPublicAccess)
-		// For all Clouds except Public, "min_tls_version" is not returned from Azure so always persist the default values for "min_tls_version".
+		// For all Clouds except Public and USGovernmentCloud, "min_tls_version" is not returned from Azure so always persist the default values for "min_tls_version".
 		// https://github.com/terraform-providers/terraform-provider-azurerm/issues/7812
 		// https://github.com/terraform-providers/terraform-provider-azurerm/issues/8083
-		if meta.(*clients.Client).Account.Environment.Name != autorestAzure.PublicCloud.Name {
+		// USGovernmentCloud "min_tls_version" allowed as of issue 9128
+		// https://github.com/terraform-providers/terraform-provider-azurerm/issues/9128
+		if meta.(*clients.Client).Account.Environment.Name != autorestAzure.PublicCloud.Name && meta.(*clients.Client).Account.Environment.Name != autorestAzure.USGovernmentCloud.Name {
```

Comment:
could we pull out env name like above:
```suggestion
		// https://github.com/terraform-providers/terraform-provider-azurerm/issues/9128
		envName := meta.(*clients.Client).Account.Environment.Name
		if  envName != autorestAzure.PublicCloud.Name && envName != autorestAzure.USGovernmentCloud.Name {
```

Reply:
Env Name pulled out to variable.